src/ErrorHandler/SimpleErrorHandler.php

```diff
@@ -51,7 +51,7 @@ protected function createResponseFromError(Throwable $error): ResponseInterface
 		$data = [
 			'status' => 'error',
 			'code' => $code,
-			'message' => $error instanceof ApiResponse ? $error->getMessage() : 'Application encountered an internal error. Please try again later.',
+			'message' => ($error instanceof ApiException && $error->getMessage() !== '') ? $error->getMessage() : 'Application encountered an internal error. Please try again later.',
```

Comment:
Without message empty check please. ApiException and it descendants should always have an message. We will enforce it inside ApiException soon.

Reply:
Anyway, with your fix will be message `'Application encountered an internal error. Please try again later.'` displayed only for exceptions which have no meaning for api layer

Reply:
Message empty check removed.